File: vault/Portal-da-Graca/Album/_Scripts/extract_lyrics.py
import os
import re
import unicodedata

# Diretórios
SCRIPT_DIR = os.path.dirname(os.path.abspath(__file__))
ALBUM_DIR = os.path.dirname(SCRIPT_DIR)
LETRAS_FILE = os.path.join(ALBUM_DIR, 'letras-completas.md')
SAIDA_DIR = os.path.join(ALBUM_DIR, 'letras')

print('Iniciando extração de letras...')

if not os.path.exists(LETRAS_FILE):
    print(f'Erro: Arquivo não encontrado - {LETRAS_FILE}')
    exit(1)

with open(LETRAS_FILE, 'r', encoding='utf-8') as f:
    conteudo = f.read()

# Expressão regular para encontrar as seções
padrao = r"## (\d{2}) — (.+?)\n\*Tema: (.+?)\*\n\n```(?:\w*\n)?([\s\S]*?)```"
matches = re.findall(padrao, conteudo)

if not os.path.exists(SAIDA_DIR):
    os.makedirs(SAIDA_DIR)

count = 0
for numero, titulo, tema, letra in matches:
    titulo = titulo.strip()
    
    # Simple slugify
    slug = unicodedata.normalize('NFKD', titulo).encode('ascii', 'ignore').decode('utf-8')
    slug = slug.lower().replace(' ', '-')
    slug = re.sub(r'[^a-z0-9\-]', '', slug)
    
    nome_arquivo = f"{numero.strip()}-{slug}.md"
    caminho_saida = os.path.join(SAIDA_DIR, nome_arquivo)
    
    novo_conteudo = f"""---
tipo: musica
numero: {numero.strip()}
titulo: "{titulo}"
tema: "{tema.strip()}"
projeto: Filhos no Amor do Pai
status: Letra
tags: [album/filhos-no-amor, lo-fi, gospel]
---

# {titulo}

```
{letra.strip()}
```
"""
    with open(caminho_saida, 'w', encoding='utf-8') as f:
        f.write(novo_conteudo)
    
    print(f"✅ Gerado: {nome_arquivo} ({titulo})")
    count += 1

print(f"\nExtração concluída! Total de letras processadas: {count}")

File: vault/Portal-da-Graca/Album/_Scripts/extract_lyrics.js
const fs = require('fs');
const path = require('path');

// Caminhos relativos ao script (que deve estar em Album/_Scripts/)
const ALBUM_DIR = path.join(__dirname, '..');
const LETRAS_FILE = path.join(ALBUM_DIR, 'letras-completas.md');
const SAIDA_DIR = path.join(ALBUM_DIR, 'letras');

console.log('Iniciando extração de letras...');

// Lê o arquivo original
if (!fs.existsSync(LETRAS_FILE)) {
    console.error(`Erro: Arquivo não encontrado - ${LETRAS_FILE}`);
    process.exit(1);
}

const conteudo = fs.readFileSync(LETRAS_FILE, 'utf8');

// Expressão regular para encontrar cada seção de letra (começa com "## " até o próximo "## " ou fim)
const mRegex = /## (\d{2}) — (.+)\n\*Tema: (.+)\*\n\n```\n([\s\S]*?)```/g;

let matches;
let Count = 0;

// Garante que a pasta de destino existe
if (!fs.existsSync(SAIDA_DIR)) {
    fs.mkdirSync(SAIDA_DIR, { recursive: true });
}

while ((matches = mRegex.exec(conteudo)) !== null) {
    const numero = matches[1].trim();
    const titulo = matches[2].trim();
    const tema = matches[3].trim();
    const letraTexto = matches[4].trim();

    // Cria um nome de arquivo seguro (slugify simples)
    const slug = titulo.toLowerCase().normalize("NFD").replace(/[\u0300-\u036f]/g, "").replace(/\s+/g, '-').replace(/[^a-z0-9\-]/g, '');
    const fileName = `${numero}-${slug}.md`;
    const outputPath = path.join(SAIDA_DIR, fileName);

    // Constrói o novo conteúdo com o YAML frontmatter
    const novoConteudo = `---
tipo: musica
numero: ${numero}
titulo: "${titulo}"
tema: "${tema}"
projeto: Filhos no Amor do Pai
status: Letra
tags: [album/filhos-no-amor, lo-fi, gospel]
---

# ${titulo}

\`\`\`
${letraTexto}
\`\`\`
`;

    // Escreve o arquivo no diretório
    fs.writeFileSync(outputPath, novoConteudo);
    console.log(`✅ Gerado: ${fileName} (${titulo})`);
    Count++;
}

console.log(`\nExtração concluída! Total de letras processadas: ${Count}`);
